agent/consul/server_test.go

```diff
@@ -1129,6 +1133,126 @@ func TestServer_RPC(t *testing.T) {
 	}
 }
 
+func TestServer_RPC_MetricsIntercept_Off(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	storage := make(map[string]float32)
+	keyMakingFunc := func(key []string) string {
+		return strings.Join(key, "+")
+	}
+
+	simpleRecorderFunc := func(key []string, val float32, labels []metrics.Label) {
+		storage[keyMakingFunc(key)] = val
```

Comment:
maybe just fail the test directly in here, we don't expect any recording to happen in here right?

Reply:
that's a good point -- I recall that folks were not inclined to do this when I was adding interceptor;

i'm slightly biased to pretend that this function does some recording and then to look at the `map` to see that the right values are in there.